pkg/controller/deployment/deployment_controller.go

```diff
@@ -438,41 +438,124 @@ func (dc *DeploymentController) syncDeployment(key string) error {
 	}
 
 	if d.Spec.Paused {
-		// TODO: Implement scaling for paused deployments.
-		// Don't take any action for paused deployment.
-		// But keep the status up-to-date.
-		// Ignore paused deployments
-		glog.V(4).Infof("Updating status only for paused deployment %s/%s", d.Namespace, d.Name)
-		return dc.syncPausedDeploymentStatus(d)
+		return dc.sync(d)
 	}
+
 	if d.Spec.RollbackTo != nil {
 		revision := d.Spec.RollbackTo.Revision
 		if _, err = dc.rollback(d, &revision); err != nil {
 			return err
 		}
 	}
 
+	if dc.isScalingEvent(d) {
+		return dc.sync(d)
```

Comment:
We definitely want requeues on RS events. Imagine a RS with replicas is
deleted just after the resync interval. The deployment is going to notice
it in the next relist if we didn't requeue.

On Tue, May 24, 2016 at 5:48 PM, Phillip Wittrock <notifications@github.com>
wrote:

> In pkg/controller/deployment/deployment_controller.go
> <https://github.com/kubernetes/kubernetes/pull/20273#discussion_r64418444>
> :
>
> >  	if d.Spec.RollbackTo != nil {
> >  		revision := d.Spec.RollbackTo.Revision
> >  		if _, err = dc.rollback(d, &revision); err != nil {
> >  			return err
> >  		}
> >  	}
> >
> > +	if dc.isScalingEvent(d) {
> > +		return dc.sync(d)
>
> Ah that makes sense. I suspect we will always want to
> re-queue-if-not-queued after scaling sync instead of relying upon updates
> to the RSs to cause the re-queue. This would be for if when the sync is
> seeing both scaling and non-scaling changes.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/20273/files/288d0835dbcd37ad101db417cf8e4395713ea64a#r64418444>
>